tests/aws/services/lambda_/test_lambda.py

```diff
@@ -1867,140 +1867,163 @@ def check_logs():
+        # This appears to have been the case at least since 2021-06.
+        # Furthermore this contradicts with what its docs on valid IAM actions for layer versions:
+        # https://docs.aws.amazon.com/lambda/latest/dg/lambda-api-permissions-ref.html#permissions-resources-layers
+
+        # Operations related to Lambda layers supported by cross account
         # - GetLayerVersion
         # - GetLayerVersionByArn
+
+        # Operations not supported by lambda layer cross account
         # - ListLayerVersions
         # - DeleteLayerVersion
         # - AddLayerVersionPermission
         # - GetLayerVersionPolicy
         # - RemoveLayerVersionPermission
 
-        # Run assertions using secondary test credentials
-
-        assert secondary_client.get_layer_version(LayerName=layer_arn, VersionNumber=1)
-
-        assert secondary_client.get_layer_version_by_arn(Arn=layer_arn + ":1")
-
-        assert secondary_client.list_layer_versions(LayerName=layer_arn)
+        layer_name = f"layer-{short_uid()}"
+        layer_arn = primary_client.publish_layer_version(
+            LayerName=layer_name, Content={"ZipFile": dummylayer}
+        )["LayerArn"]
 
-        assert secondary_client.add_layer_version_permission(
+        layer_statement_id = f"stm-{short_uid()}"
+        primary_client.add_layer_version_permission(
             LayerName=layer_arn,
-            VersionNumber=1,
+            StatementId=layer_statement_id,
+            Principal=secondary_account_id,
             Action="lambda:GetLayerVersion",
-            Principal="*",
-            StatementId="s1",
-        )
-
-        assert secondary_client.get_layer_version_policy(LayerName=layer_arn, VersionNumber=1)
-
-        assert secondary_client.remove_layer_version_permission(
-            LayerName=layer_arn, VersionNumber=1, StatementId="s1"
+            VersionNumber=1,
         )
 
-        assert secondary_client.delete_layer_version(LayerName=layer_arn, VersionNumber=1)
+        yield layer_arn
 
-        # Operations related to functions.
-        # See: https://docs.aws.amazon.com/lambda/latest/dg/access-control-resource-based.html#permissions-resource-xaccountinvoke
-        #
-        # - Invoke
-        # - GetFunction
-        # - GetFunctionConfiguration
-        # - UpdateFunctionCode (NOT TESTED)
-        # - DeleteFunction
-        # - PublishVersion
-        # - ListVersionsByFunction
-        # - CreateAlias
-        # - GetAlias
-        # - ListAliases
-        # - UpdateAlias
-        # - DeleteAlias
-        # - GetPolicy (NOT TESTED)
-        # - PutFunctionConcurrency
-        # - DeleteFunctionConcurrency
-        # - ListTags
-        # - TagResource
-        # - UntagResource
+    @markers.aws.validated
+    def test_get_lambda_layer(self, secondary_client, created_layer):
+        layer_arn = created_layer
```

Comment:
Nit: Is there a more specific reason for this assignments? Otherwise we could just directly use `created_layer` (could even rename to `created_layer_arn` to make it clearer) directly, and avoid the additional assignment.

Reply:
Done :)